File: CCC_2020/S1/main.cpp
#include <bits/stdc++.h>
using namespace std;

struct point{
    int time, position;
    bool operator<(const point& p){
        return time < p.time;
    }
};

int n;
double fastest = 0;
point points[100000];//problem description's restrictions
int main(){
    std::cin >> n;
    for(int i = 0;i < n;i++){
        std::cin >> points[i].time;
        std::cin >> points[i].position;
    }
    sort(points, points + n);
    for(int i = 1;i < n;i++){
        double time_dif = points[i].time - points[i - 1].time;
        double dist_dif = points[i].position - points[i - 1].position;
        double speed = std::abs(dist_dif/time_dif);
        if(speed > fastest)fastest = speed;
    }
    printf("%f", fastest);
    return 0;
}

File: CCC_2020/S1/main.py
n = int(input())
speed_dist = []
for i in range(n):
    temp_arr = []
    s = input()
    temp_arr.append(int(s.split(" ")[0]))
    temp_arr.append(int(s.split(" ")[1]))
    speed_dist.append(temp_arr)

speed_dist.sort()

x = -1
for i in range(1, n):
    time_diff = speed_dist[i][0] - speed_dist[i - 1][0]
    dist_diff = abs(speed_dist[i][1] - speed_dist[i - 1][1])
    speed = dist_diff/time_diff
    if speed > x:
        x = speed
print(x)